processor/src/main/kotlin/permissions/dispatcher/processor/util/Extensions.kt

```diff
@@ -122,15 +122,19 @@ fun FileSpec.Builder.addTypes(types: List<TypeSpec>): FileSpec.Builder {
 fun TypeName.correctJavaTypeToKotlinType(): TypeName {
     return if (this is ParameterizedTypeName) {
         val typeArguments = this.typeArguments.map { it.correctJavaTypeToKotlinType() }.toTypedArray()
-        return this.rawType.parameterizedBy(*typeArguments)
+        val rawType = ClassName.bestGuess(this.rawType.correctJavaTypeToKotlinType().toString())
+        return rawType.parameterizedBy(*typeArguments)
     } else when (toString()) {
         "java.lang.Byte" -> ClassName("kotlin", "Byte")
         "java.lang.Double" -> ClassName("kotlin", "Double")
         "java.lang.Object" -> ClassName("kotlin", "Any")
         "java.lang.String" -> ClassName("kotlin", "String")
+        "java.util.Set" -> ClassName("kotlin.collections", "MutableSet")
```

Comment:
not `kotlin.collections.Set`? 👀

Reply:
`java.util.Set` is more like `kotlin.collections.MutableSet`.

If you declare method like
```kotlin
@NeedsPermission(CAMERA, RECORD_AUDIO)
fun doWork(values: kotlin.collections.MutableSet<kotlin.String>) {
    ...
}
```
The generated code won't be compilable, because it won't be able to pass `kotlin.collections.Set` into method with `kotlin.collections.MutableSet` parameter.
And vice versa `kotlin.collections.MutableSet` fits `kotlin.collections.Set`.